src/v/bytes/bytes.h

```diff
@@ -134,3 +134,25 @@ bytes_type_eq::operator()(const bytes& lhs, const iobuf& rhs) const {
     }
     return true;
 }
+
+inline bytes operator^(bytes_view a, bytes_view b) {
+    if (unlikely(a.size() != b.size())) {
+        throw std::runtime_error(
+          "Cannot compute xor for different size byte strings");
+    }
+    bytes res(bytes::initialized_later{}, a.size());
+    std::transform(
+      a.cbegin(), a.cend(), b.cbegin(), res.begin(), [](uint8_t a, uint8_t b) {
+          return a ^ b;
```

Comment:
hot

Reply:
question, why not use `std::transform ... std::bit_xor<uint8_t>()` vs the lambda

Reply:
ah thx. i never remember those objects exist, despite using it 8 lines below this in the same diff. ill change this before merging.